src/pilot/utils.patch

```diff
@@ -41,39 +41,86 @@
-+                 wait_for_idrac=True):
++                 wait_for_idrac=True,
++                 by_name=True):
      """Generically handles setting various types of settings on the iDRAC
  
      This method pulls the current list of settings from the iDRAC then compares
-@@ -339,6 +343,9 @@
+@@ -339,6 +344,11 @@
                             attribute.name will be used.
      :parm include_commit_required: Indicates if the deprecated commit_required
                                     should be returned in the result.
 +    :param wait_for_idrac: indicates whether or not to wait for the
 +                           iDRAC to be ready to accept commands before issuing
 +                           the command
++    :param by_name: Controls whether returned dictionary uses RAID
++                    attribute name or instance_id as key
      :returns: a dictionary containing:
               - The commit_required key with a boolean value indicating
                 whether a config job must be created for the values to be
-@@ -361,7 +368,8 @@
+@@ -360,8 +370,9 @@
+     :raises: InvalidParameterValue on invalid new setting
      """
  
-     current_settings = list_settings(client, namespaces, by_name=True,
+-    current_settings = list_settings(client, namespaces, by_name=True,
 -                                     name_formatter=name_formatter)
++    current_settings = list_settings(client, namespaces, by_name=by_name,
 +                                     name_formatter=name_formatter,
 +                                     wait_for_idrac=wait_for_idrac)
  
      unknown_keys = set(new_settings) - set(current_settings)
      if unknown_keys:
-@@ -428,7 +436,8 @@
+@@ -376,11 +387,23 @@
+     candidates = set(new_settings)
+ 
+     for attr in candidates:
+-        if str(new_settings[attr]) == str(
+-                current_settings[attr].current_value):
+-            unchanged_attribs.append(attr)
+-        elif current_settings[attr].read_only:
++        # There are RAID settings that can have multiple values,
++        # however these are all read-only attributes.
++        # Filter out all read-only attributes first so that we exclude
++        # these settings from further consideration
++        current_setting_value = current_settings[attr].current_value
++        if type(current_setting_value) is list:
++            current_setting_value = current_setting_value[0]
++
++        unchanged_attribute = str(new_settings[attr]) == str(
++            current_setting_value)
++
++        # check if read-only attribute is unchanged
++        if current_settings[attr].read_only and not unchanged_attribute:
+             read_only_keys.append(attr)
++
++        if unchanged_attribute:
++            unchanged_attribs.append(attr)
+         else:
+             validation_msg = current_settings[attr].validate(
+                 new_settings[attr])
+@@ -425,8 +448,22 @@
+                   'AttributeName': attrib_names,
```

Comment:
It looks to me like the AttributeName line above was removed here:
https://review.opendev.org/#/c/715172/5/dracclient/utils.py

Reply:
Thanks for pointing out..i have updated in next commit.